byterun/sys.c

```diff
@@ -175,15 +176,27 @@ CAMLprim value caml_sys_open(value path, value vflags, value vperm)
 {
   CAMLparam3(path, vflags, vperm);
   int fd, flags, perm;
+#ifdef _WIN32
+  wchar_t * p;
+#else
   char * p;
+#endif
 
   caml_sys_check_path(path);
+#ifdef _WIN32
+  p = caml_stat_strdup_to_utf16(String_val(path));
+#else
   p = caml_stat_strdup(String_val(path));
+#endif
```

Comment:
I like the shorter `caml_stat_os_strdup` for `caml_stat_strdup_to_utf16`, but then what do we use for `caml_stat_strdup_of_utf16` ?